store/src/java/com/zimbra/cs/service/util/S3BucketGlobalConfigUtil.java

```diff
@@ -0,0 +1,485 @@
+    			JSONArray gs3bucketsJA = gescLdapJO.getJSONArray(GlobalExternalStoreConfigConstants.E_GLOBAL_S3_BUCKET_CONFIGURATIONS);
+    			if(gs3bucketsJA != null) {
+	    			for(int i = 0; i < gs3bucketsJA.length(); i++) {
+	    				JSONObject gs3bucketConfJO = gs3bucketsJA.getJSONObject(i);
+	    				boolean bucketIsActive = ((String) gs3bucketConfJO.get(GlobalExternalStoreConfigConstants.A_S3_BUCKET_STATUS)).equalsIgnoreCase("ACTIVE");
+	    				if(bucketIsActive) {
+	    					GlobalS3BucketConfiguration gs3bucketConf = new GlobalS3BucketConfiguration();
+	    					for (Iterator it = gs3bucketConfJO.keys(); it.hasNext(); ) {
+	    						Object fieldObject = it.next();
+	    						if(fieldObject instanceof String) {
+	    							String field = String.valueOf(fieldObject);
+	    							String value = (String) gs3bucketConfJO.get(field);
+	    							setValueByReflection(gs3bucketConf, field, value);
+	    						}
+	    					}
+	    					gesc.addGlobalS3BucketConfiguration(gs3bucketConf);
+	    				}
+	    			}
+    			}
+    		}
+		
+		} catch (JSONException e) {
+			ZimbraLog.mailbox.warn("Failed to parse ldap global_external_store_config parameter", e);
+			throw ServiceException.FAILURE("Failed to parse ldap global_external_store_config parameter", null); 
+		}
+		
+		return gesc;
+    	
+    }
+    
+    /**
+     * Parse the Attr list inside of request element to a GlobalS3BucketConfiguration object
+     *
+     * @param request
+     * @return GlobalS3BucketConfiguration
+     * @throws ServiceException
+     */
+    public static GlobalS3BucketConfiguration parseCreateRequestToGlobalS3BucketConfig(CreateS3BucketConfigRequest request) throws ServiceException {
+    	
+    	GlobalS3BucketConfiguration gs3bc = new GlobalS3BucketConfiguration();
+    		
+		for (Attr attr : request.getAttrs()) {
+			setValueByReflection(gs3bc, attr.getKey(), attr.getValue());
+		}
+		
+		return gs3bc;
+    	
+    }
+    
+    /**
+     * Parse the Attr list inside of request element to a GlobalS3BucketConfiguration object
+     *
+     * @param request
+     * @return GlobalS3BucketConfiguration
+     * @throws ServiceException
+     */
+    public static GlobalS3BucketConfiguration parseDeleteRequestToGlobalS3BucketConfig(DeleteS3BucketConfigRequest request) throws ServiceException {
+    	
+    	GlobalS3BucketConfiguration gs3bc = new GlobalS3BucketConfiguration();
+    		
+		for (Attr attr : request.getAttrs()) {
+			setValueByReflection(gs3bc, attr.getKey(), attr.getValue());
+		}
+		
+		return gs3bc;
+    	
+    }
+    
+    /**
+     * Given an object, set the value for a specific field using Java Reflection API.
+     *
+     * @param object
+     * @param fieldStr
+     * @param valueStr
+     * @return
+     * @throws ServiceException
+     */
+    private static void setValueByReflection(Object object, String fieldStr, String valueStr) throws ServiceException {
+		
+		boolean breakBucle = false;
```

Comment:
this var breakBucle name isnt giving details about its usage , can  we rename it